src/marqo/tensor_search/tensor_search.py

```diff
@@ -1238,3 +1239,22 @@ def _get_model_properties(index_info):
                 f"Please provide model_properties if the model is a custom model and is not supported by default")
 
     return model_properties
+
+def get_loaded_models() -> dict:
+    available_models = s2_inference.get_available_models()
+    message = {
+        'models' : [
+            {"model_name": ix} for ix in available_models
+        ]
+    }
+    return message
+def eject_model(model_name: str, device: str) -> dict:
+    try:
+       result = s2_inference.eject_model(model_name, device)
+    except s2_inference_errors.ModelNotLoadedError as e:
+        raise errors.ModelNotLoadedError(message=str(e))
+    return result
+def get_cuda_info() -> dict:
+    return {"device": "cuda",
+            "memory_usage": f"{round(torch.cuda.memory_allocated() / 1024**3, 1)} GiB",
+            "total_device_memory": f"{round(torch.cuda.get_device_properties(0).total_memory/ 1024**3, 1)} GiB"}
```

Comment:
In this call `torch.cuda.get_device_properties(0)` what does the 0 refer to? Is that the index of the device (for example `cuda:0`)?

Reply:
We now support multi-gpu information:

On 1-gpu device instance:
{'cuda_usage_info': [{'device_id': 0, 'device_name': 'Tesla V100-SXM2-16GB', 'memory_used': '0.0 GiB', 'total_memory': '15.8 GiB'}]}
​
On 4-gpu devices instance:
{'cuda_usage_info': [{'device_id': 0, 'device_name': 'Tesla V100-SXM2-16GB', 'memory_used': '0.0 GiB', 'total_memory': '15.8 GiB'}, {'device_id': 1, 'device_name': 'Tesla V100-SXM2-16GB', 'memory_used': '0.0 GiB', 'total_memory': '15.8 GiB'}, {'device_id': 2, 'device_name': 'Tesla V100-SXM2-16GB', 'memory_used': '0.0 GiB', 'total_memory': '15.8 GiB'}, {'device_id': 3, 'device_name': 'Tesla V100-SXM2-16GB', 'memory_used': '0.0 GiB', 'total_memory': '15.8 GiB'}]}